cs/src/core/Index/FASTER/FASTER.cs

```diff
@@ -96,10 +98,30 @@ public FasterKV(long size, Functions functions, LogSettings logSettings, Checkpo
             CopyReadsToTail = logSettings.CopyReadsToTail;
             this.functions = functions;
 
+            if (logSettings.ReadCacheSizeBits > 0)
+            {
+                CopyReadsToTail = false;
+                UseReadCache = true;
+            }
+
             if (Utility.IsBlittable<Key>() && Utility.IsBlittable<Value>())
+            {
                 hlog = new BlittableAllocator<Key, Value>(logSettings, this.comparer);
+                if (UseReadCache)
+                {
+                    readcache = new BlittableAllocator<Key, Value>(logSettings, this.comparer);
+                    readcache.Initialize();
+                }
+            }
             else
+            {
                 hlog = new GenericAllocator<Key, Value>(logSettings, serializerSettings, this.comparer);
+                if (UseReadCache)
+                {
+                    readcache = new GenericAllocator<Key, Value>(logSettings, serializerSettings, this.comparer);
+                    readcache.Initialize();
+                }
+            }
```

Comment:
If the entire data set ends up fitting in the readcache, is there a way to "shrink" the readcache to minimize the required footprint?

Reply:
I have made it possible to call ShiftHeadAddress on the read cache. The testcase shows how to do this.
            // Evict the read cache entirely
            fht.ReadCache.ShiftHeadAddress(fht.ReadCache.GetTailAddress());